roles/kubernetes/kubeadm/tasks/main.yml

```diff
@@ -29,14 +29,15 @@
   delegate_to: "{{ groups['kube_control_plane'][0] }}"
   run_once: true
 
-- name: Calculate kubeadm CA cert hash
-  shell: |
-    set -o pipefail && openssl x509 -pubkey -in {{ kube_cert_dir }}/ca.crt | \
-    openssl {% if 'RSA' in kube_asymmetric_encryption_algorithm %}rsa{% elif 'ECDSA' in kube_asymmetric_encryption_algorithm %}ec{% else %}rsa{% endif %} -pubin -outform der 2>/dev/null | \
-    openssl dgst -sha256 -hex | sed 's/^.* //'
-  args:
-    executable: /bin/bash
-  register: kubeadm_ca_hash
+- name: Extract public key from CA certificate
+  community.crypto.openssl_publickey_info:
+    path: "{{ kube_cert_dir }}/ca.crt"
+    type: x509
+  register: publickey_info
```

Comment:
The problem with that is that community.crypto module requires to install python cryptography **managed host**, which is not that easy to do -> see #11158 
An alternative to that, I think, is to fetch either the public key or the whole cert (but not the private key, ofc) and use the community.crypto filters ( for instance the [x509_info filter](https://docs.ansible.com/ansible/latest/collections/community/crypto/x509_certificate_info_filter.html#ansible-collections-community-crypto-x509-certificate-info-filter) ).
Since filter runs on the ansible control node, this only requires to install cryptography on the control node, which is much easier.